### PR TITLE
GStreamer Major update -> 1.20.0

### PR DESCRIPTION
**Pull request by BFamzz**
GStreamer and other associated libraries (plugins, rtsp-server, python, etc.) have been updated to the latest major release -> 1.20.0

### CODE REVIEW

**Comment by lazka** [comment]
Let us know if you need any help

File: mingw-w64-gst-editing-services/PKGBUILD [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Maintainer: Lubosz Sarnecki <lubosz@gmail.com>
+#Maintainer: Lubosz Sarnecki <lubosz@gmail.com>
```
Comment (filnet) [inline mingw-w64-gst-editing-services/PKGBUILD:1]: keyboard slippage ?

Comment (BFamzz) [reply mingw-w64-gst-editing-services/PKGBUILD:1]: @filnet yea. Sorry my bad. I'll fix it.

**Comment by BFamzz** [comment]
> Let us know if you need any help

Alright @lazka. Thanks. 

The first build failed with an error from the patch file [0001-meson-fix-vulkan-detection.patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-gst-plugins-bad/0001-meson-fix-vulkan-detection.patch). 

Still looking at what could have caused it.

File: mingw-w64-gst-editing-services/PKGBUILD (hunk 1 above)
Comment (BFamzz) [reply mingw-w64-gst-editing-services/PKGBUILD]: It's been fixed.

**Comment by lazka** [comment]
```
2022-02-04T17:49:32.0722016Z ==> Starting build()...
2022-02-04T17:49:32.8411684Z The Meson build system
2022-02-04T17:49:32.8412159Z Version: 0.61.1
2022-02-04T17:49:32.8412970Z Source dir: C:/_/mingw-w64-gst-plugins-bad/src/gst-plugins-bad-1.20.0
2022-02-04T17:49:32.8413585Z Build dir: C:/_/mingw-w64-gst-plugins-bad/src/build-x86_64-w64-mingw32
2022-02-04T17:49:32.8413953Z Build type: native build
2022-02-04T17:49:32.8414176Z 
2022-02-04T17:49:32.8414641Z ../gst-plugins-bad-1.20.0/meson.build:1:0: ERROR: Unknown options: "libmms, ofa"
2022-02-04T17:49:32.8414942Z 
2022-02-04T17:49:32.8415363Z A full log can be found at C:/_/mingw-w64-gst-plugins-bad/src/build-x86_64-w64-mingw32/meson-logs/meson-log.txt
```

libmms and ofa have been removed according to https://gstreamer.freedesktop.org/releases/1.20 , so the build options can simply be removed from the PKGBUILD.

**Comment by BFamzz** [comment]
@lazka @Biswa96 help required please. 

The header file `vulkan/vulkan_win32.h` was not found during the build process.

Excerpt from the meson.build file

```
if host_system == 'windows'
  gdi_dep = cc.find_library('gdi32', required : get_option('vulkan'))

  # Cannot use internal dependency object with cc.has_header()
  if gdi_dep.found() and cc.has_header('vulkan/vulkan_win32.h', args: '-I' + vulkan_inc_dir)
    vulkan_priv_sources += ['win32/gstvkwindow_win32.c']
    optional_deps += [gdi_dep]
    vulkan_windowing = true
    vulkan_conf.set10('GST_VULKAN_HAVE_WINDOW_WIN32', 1)
    enabled_vulkan_winsys += ['win32']
  endif
endif
```

Any workaround ideas please?

Thanks.

**Comment by lazka** [comment]
I'll have a look

**Comment by lazka** [comment]
> that change is strange ?

hm?

**Comment by filnet** [comment]
> GCC errored out complaining that the function is used before being declared.

Ok, the change makes sense now.

**Comment by lazka** [comment]
thanks!

**Pull request merged**